services/bin/bambi_service

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash -e
+#
+# BAMBI backend worker
+#
+SCRIPT_ROOT=$(realpath $(dirname $0))
+UDMI_ROOT=$(realpath "$SCRIPT_ROOT"/../..)
+
+source "$UDMI_ROOT"/etc/shell_common.sh
+
+export PS4='+(${BASH_SOURCE}:${LINENO}): ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
+
+LOCAL=
+[[ ${1:-} == "--local" ]] && LOCAL=1 && shift
+
+NO_CHECK=
+[[ ${1:-} == "--no-check" ]] && NO_CHECK=1 && shift
+
+if [[ -n ${UDMI_NAMESPACE:-} ]]; then
+    UDMI_PREFIX="${UDMI_NAMESPACE}~"
+    echo Using UDMI_PREFIX "$UDMI_PREFIX"
+else
+    UDMI_PREFIX=
+fi
+
+# shellcheck disable=SC2155
+export GCP_PROJECT=$(gcloud config get-value project)
+echo Using GCP_PROJECT "$GCP_PROJECT"
+
+if [[ -z $LOCAL ]]; then
+    GCP_SERVICE_ACCOUNT=$(curl -s -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/instance/service-accounts/default/email)
+    echo Configured with GCP service account "$GCP_SERVICE_ACCOUNT"
+    gcloud config set account "$GCP_SERVICE_ACCOUNT"
+    git config --global user.email "$GCP_SERVICE_ACCOUNT"
+    git config --global user.name "BAMBI backend runner ${UDMI_PREFIX#\~}"
+else
+  [[ ! -d $UDMI_ROOT/services/src ]] ||
+      up_to_date "$SERVICES_JAR" "$UDMI_ROOT"/services/src ||
+      "$UDMI_ROOT"/services/bin/build
+fi
+
+BAMBI_REQUESTS_TOPIC=${UDMI_PREFIX}bambi-requests
+BAMBI_REQUESTS_SUBSCRIPTION=${UDMI_PREFIX}bambi-requests
+SITE_MODEL_CLONE_DIR=${UDMI_ROOT}/var/udmi/sites
+
+SLEEP_INTERVAL=0.2
+
+if [[ -z $NO_CHECK ]]; then
+    echo Checking existing topics...
+    gcloud pubsub subscriptions list | grep -F name: || true
+    existing_topic=$(gcloud pubsub topics list | grep -F "${BAMBI_REQUESTS_TOPIC}" || true)
+    if [[ -n $existing_topic ]]; then
+        echo Topic "${BAMBI_REQUESTS_TOPIC}" exists
+    else
+        echo Creating topic "${BAMBI_REQUESTS_TOPIC}"
+        gcloud pubsub topics create "${BAMBI_REQUESTS_TOPIC}"
+    fi
+
+    echo Checking existing subscriptions...
+    gcloud pubsub subscriptions list | grep -F name: || true
+    existing=$(gcloud pubsub subscriptions list | grep -F "${BAMBI_REQUESTS_SUBSCRIPTION}" || true)
+    if [[ -n $existing ]]; then
+        echo Using existing subscription "${BAMBI_REQUESTS_SUBSCRIPTION}"
+    else
+        gcloud pubsub subscriptions create "${BAMBI_REQUESTS_SUBSCRIPTION}" --topic="${BAMBI_REQUESTS_TOPIC}" --expiration-period=24h
+    fi
+fi
+
+echo Polling for messages...
+
+while true; do
+    PULL_OUTPUT=$(gcloud pubsub subscriptions pull \
+        "${BAMBI_REQUESTS_SUBSCRIPTION}" \
+        --auto-ack \
```

Comment:
to consider for the future, what is the failure mechanism if there is an error? auto-ack ackknowledges the message, so if there is an error the request is lost.